core-plugins/src/main/java/co/cask/hydrator/plugin/batch/aggregator/RowDenormalizerConfig.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.plugin.batch.aggregator;
+
+import co.cask.cdap.api.annotation.Description;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Splitter;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Config for RowDenormalizer Aggregator Plugin.
+ */
+public class RowDenormalizerConfig extends AggregatorConfig {
+
+  @Description("Name of the column in the input record which will be used to group the raw data. For Example, " +
+    "KeyField.")
+  private final String keyField;
+
+  @Description("Name of the column in the input record which contains the names of output schema columns. " +
+    "For example, " +
+    "input records have columns 'KeyField', 'FieldName', 'FieldValue'. " +
+    "'FieldName' contains 'FirstName', 'LastName', 'Address'. " +
+    "So the output record will have column names as 'FirstName', 'LastName', 'Address'.")
+  private final String fieldName;
+
+  @Description("Name of the column in the input record which contains the values for output schema columns. " +
+    "For example, " +
+    "input records have columns 'KeyField', 'FieldName', 'FieldValue' " +
+    "and the 'FieldValue' column contains 'John', 'Wagh', 'NE Lakeside'. " +
+    "So the output record will have values for columns 'FirstName', 'LastName', 'Address' as 'John', 'Wagh', 'NE " +
+    "Lakeside' respectively.")
+  private final String fieldValue;
+
+  @Description("List of the output fields with its alias to be included in denormalized output. This is a comma " +
+    "separated list of key-value pairs, where key-value pairs are separated by a colon ':'. For example, " +
+    "Firstname:fname,Lastname:lname,Address:addr")
```

Comment:
can we avoid specifying the output fields and their alias in the same property? For example, make this purely a list of output fields, and have an addition property called fieldAliases that is a list of key-value pairs. I think this will be much more clear.